tests/end_to_end_test/restore_test.go

```diff
@@ -488,6 +493,271 @@ func TestRestoreSnapshotOfSingleFile(t *testing.T) {
 	verifyFileMode(t, filepath.Join(restoreDir, "restored-5"), defaultRestoredFilePermission)
 }
 
+func TestSnapshotSparseRestore(t *testing.T) {
+	t.Parallel()
+
+	if !(runtime.GOOS == linuxOSName && runtime.GOARCH == amdArchName) {
+		t.Skip("Sparse file behavior not guaranteed for this OS/Arch pair")
```

Comment:
can you add more comments here?

Reply:
Done. Also moved this to `testutil.TestSkipUnlessLinux()`